amethyst_renderer/src/pipe/stage.rs

```diff
@@ -37,51 +37,78 @@ impl Stage {
         self.enabled
     }
 
-    /// Get passes of the stage
-    pub fn passes(&self) -> &[Pass] {
-        self.passes.as_slice()
-    }
+    /// Applies all passes within this stage.
+    pub fn apply(&self, encoders: &mut [Encoder], scene: &Scene) {
+        use rayon::prelude::*;
+
+        let mut encs = encoders;
+        self.clear_color.map(|c| self.target.clear_color(&mut encs[0], c));
+        self.clear_depth.map(|d| self.target.clear_depth_stencil(&mut encs[0], d));
+
+        for pass in self.passes.iter() {
+            let mut models = scene.par_chunks_models(4);
```

Comment:
4 is a good number. But constant is really bad here.
Use `encoders.len() / self.passes.len()`.
Or on machines with less then 4 cores there will be panic 3 lines below.